implementations/rust/ockam/ockam_vault/src/verifier_impl.rs

```diff
@@ -2,26 +2,37 @@ use crate::software_vault::SoftwareVault;
 use crate::xeddsa::XEddsaVerifier;
 use crate::VaultError;
 use arrayref::array_ref;
-use ockam_vault_core::{PublicKey, Verifier, CURVE25519_PUBLIC_LENGTH};
+use ockam_vault_core::{Buffer, PublicKey, Verifier, CURVE25519_PUBLIC_LENGTH};
+use signature_bbs_plus::Signature;
+use signature_bbs_plus::MessageGenerators;
+use signature_core::lib::Message;
 
 impl Verifier for SoftwareVault {
     /// Verify signature with xeddsa algorithm. Only curve25519 is supported.
     fn verify(
         &mut self,
-        signature: &[u8; 64],
+        signature: Buffer<u8>,
         public_key: &PublicKey,
         data: &[u8],
     ) -> ockam_core::Result<bool> {
         // TODO: Add public key type
         if public_key.as_ref().len() == CURVE25519_PUBLIC_LENGTH {
+            let signature_array = array_ref!(signature, 0, 64);
```

Comment:
Should not panic if length is invalid

Reply:
Ok. Now I check if signature length equals to 64 bit. Should not panic now.